lib/ddtrace/tagging/analytics.rb

```diff
@@ -0,0 +1,22 @@
+require 'ddtrace/ext/analytics'
+require 'ddtrace/analytics'
+
+module Datadog
+  module Tagging
+    # Defines analytics tagging behavior
+    module Analytics
+      def set_tag(key, value)
+        case key
+        when Ext::Analytics::TAG_ENABLED
+          # If true, set rate to 1.0, otherwise set 0.0.
+          value = value == true ? Ext::Analytics::DEFAULT_SAMPLE_RATE : 0.0
+          Datadog::Analytics.set_sample_rate(self, value)
+        when Ext::Analytics::TAG_SAMPLE_RATE
+          Datadog::Analytics.set_sample_rate(self, value)
+        else
+          super if defined?(super)
+        end
+      end
+    end
+  end
+end
```

Comment:
Are `Datadog::Tagging::Analytics` and `Datadog::Tagging::ManualTracing` used independently of `Metadata` anywhere?

If not, I would suggest actually merging them back into `Metadata`: With the current arrangement, when someone that wants to understand and follow the current behavior of `Span#set_tag` they need to find 3 different methods across 3 different files (none of which is actually `span.rb`), which is really hard to follow.